tomviz/ModuleContourWidget.h

```diff
@@ -0,0 +1,94 @@
+/******************************************************************************
+
+  This source file is part of the tomviz project.
+
+  Copyright Kitware, Inc.
+
+  This source code is released under the New BSD License, (the "License").
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+******************************************************************************/
+#ifndef tomvizModuleContourWidget_h
+#define tomvizModuleContourWidget_h
+
+#include <memory>
+
+#include <QWidget>
+
+/**
+ * \brief UI layer of ModuleContour.
+ *
+ * Signals are forwarded to  ModuleContour or the proxies. This class is
+ * intended
+ * to contain only logic related to UI actions.
+ */
+
+namespace Ui {
+class ModuleContourWidget;
+class LightingParametersForm;
+}
+
+class QComboBox;
+
+class pqPropertyLinks;
+class vtkSMProxy;
+class vtkSMSourceProxy;
+
+namespace tomviz {
+
+class ModuleContourWidget : public QWidget
+{
+  Q_OBJECT
+
+public:
+  ModuleContourWidget(QWidget* parent_ = nullptr);
+  ~ModuleContourWidget() = default;
+
+  //@{
+  /**
+   * UI update methods. The actual model state is stored in ModuleContour for
+   * these parameters, so the UI needs to be updated if the state changes or
+   * when constructing the UI.
+   */
+  void setUseSolidColor(const bool useSolid);
+  //@}
+
+  /**
+   * Link proxy properties to UI.
+   */
+  void addPropertyLinks(pqPropertyLinks& links,
+                        vtkSMProxy* contourRepresentation,
+                        vtkSMSourceProxy* contourFilter);
+
+  /**
+   * Expose 'ColorBy' combo box to be populated by the source.
+   */
+  QComboBox* getColorByComboBox();
+
+signals:
+  //@{
+  /**
+   * Forwarded signals.
+   */
+  void specularPowerChanged(const double value);
+  void useSolidColor(const bool value);
+  /**
+   * All proxy properties should use this signal.
+   */
+  void propertyChanged();
+  //@}
+
+private:
+  ModuleContourWidget(const ModuleContourWidget&) = delete;
+  void operator=(const ModuleContourWidget&) = delete;
+
+  std::shared_ptr<Ui::ModuleContourWidget> m_ui;
```

Comment:
I can do a quick pass, but these should just be QScopedPointer, they are not shared and I would prefer to be explicit that they are not intended to be.